version: '3'

volumes:
  django_test_local_postgres_data: {}
  django_test_local_postgres_data_backups: {}

services:
  django:
    build:
      context: .
      dockerfile: ./compose/local/django/Dockerfile
    image: django_test_local_django
    container_name: django_test_local_django
    restart: always
    depends_on:
      - postgres
    volumes:
      - ~/.gitconfig:/root/.gitconfig
      - .:/app:z
    env_file:
      - ./.envs/.local/.django
      - ./.envs/.local/.postgres
    ports:
      - "8000:8000"
    command: /start

  postgres:
    build:
      context: .
      dockerfile: ./compose/production/postgres/Dockerfile
    image: django_test_production_postgres
    container_name: django_test_local_postgres
    ports:
      - "5433:5432"
    volumes:
      - django_test_local_postgres_data:/var/lib/postgresql/data
      - django_test_local_postgres_data_backups:/backups
    env_file:
      - ./.envs/.local/.postgres

  docs:
    image: django_test_local_docs
    container_name: django_test_local_docs
    build:
      context: .
      dockerfile: ./compose/local/docs/Dockerfile
    env_file:
      - ./.envs/.local/.django
    volumes:
      - ./docs:/docs:z
      - ./config:/app/config:z
      - ./django_test:/app/django_test:z
    ports:
      - "9000:9000"
    command: /start-docs
